includes/rest-api/class-sensei-rest-api-messages-controller.php

```diff
@@ -0,0 +1,135 @@
+		// This filter is needed in order for students to be able to see their own messages only.
+		add_filter( 'rest_sensei_message_query', array( $this, 'exclude_others_comments' ), 10, 2 );
+
+		register_rest_field(
+			'sensei_message',
+			'sender',
+			array(
+				'get_callback' => function( $object ) {
+					return get_post_meta( $object['id'], '_sender', true );
+				},
+				'context'      => array( 'view' ),
+				'schema'       => array(
+					'description' => __( 'The username of the message\'s sender.' ),
+					'type'        => 'string',
+				),
+			)
+		);
+
+		register_rest_field(
+			'sensei_message',
+			'displayed_title',
+			array(
+				'get_callback' => function( $object ) {
+					return Sensei_Messages::get_the_message_title( $object['id'] );
+				},
+				'context'      => array( 'view' ),
+				'schema'       => array(
+					'description' => __( 'The message\'s title.' ),
+					'type'        => 'string',
+				),
+			)
+		);
+
+		register_rest_field(
+			'sensei_message',
+			'displayed_date',
+			array(
+				'get_callback' => function() {
+					return get_the_date();
+				},
+				'context'      => array( 'view' ),
+				'schema'       => array(
+					'description' => __( 'The formatted date that the message was created.' ),
+					'type'        => 'string',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Overrides get_item_schema to add the 'excerpt'. The 'exceprt' is also returned in WP_REST_Posts_Controller but
+	 * only when the post type supports the 'excerpt' feature which we cannot add since it is going to have side effects.
+	 *
+	 * @since 2.3.0
+	 *
+	 * @return array The schema.
+	 */
+	public function get_item_schema() {
+		$schema = parent::get_item_schema();
+
+		$schema['properties']['excerpt'] = array(
+			'description' => __( 'The excerpt for the object.' ),
+			'type'        => 'object',
+			'context'     => array( 'view' ),
+			'properties'  => array(
+				'raw'       => array(
+					'description' => __( 'Excerpt for the object, as it exists in the database.' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+				),
+				'rendered'  => array(
+					'description' => __( 'HTML excerpt for the object, transformed for display.' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'protected' => array(
+					'description' => __( 'Whether the excerpt is protected with a password.' ),
+					'type'        => 'boolean',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		return $schema;
+	}
+
+
+	/**
+	 * This method adds filters to the sensei message query to restrict students to see their own messages only.
+	 *
+	 * @since 2.3.0
+	 * @access private
+	 *
+	 * @param array $args The query args.
+	 * @return array The modified query args.
+	 */
+	public function exclude_others_comments( $args ) {
+
+		if ( current_user_can( 'moderate_comments' ) || current_user_can( 'read_private_sensei_messages' ) ) {
```

Comment:
I'm still not quite sure about this. On one hand, it makes sense for the admin to be able to see all messages. On the other hand, the shortcode that we're replacing only displays the messages that were created by the current user.

Maybe there should be a parameter specifying that I want "my" messages, or "all" messages? Or if we want this endpoint to only return "my" messages, we could rename it as such? `/wp/v2/sensei-my-messages` or something like that?

Reply:
Yeah I think that we definetely need to support the exact functiolity that the shortcode did and if we add an extra it should be optional. I'll add an include='all' argument which is going to be available for users with appropriate rights only.